### PR TITLE
Network acl rules support

### PR DESCRIPTION
**Pull request by simongodard**

### CODE REVIEW

File: services/cloudca/network_acl_rule.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+//Get network acl rule with the specified id for the current environment
+func (networkAclRuleApi *NetworkAclRuleApi) Get(id string) (*NetworkAclRule, error) {
+	data, err := networkAclRuleApi.entityService.Get(id, map[string]string{})
+	if err != nil {
+		return nil, err
+	}
+	return parseNetworkAclRule(data), nil
+}
+
+func (api NetworkAclRuleApi) ListWithOptions(options map[string]string) ([]NetworkAclRule, error) {
```
Comment (prollynomial) [inline services/cloudca/network_acl_rule.go]: Isn't `api` usually a pointer? I see it is for some (`Get`, `Create`), but not `List` and `ListWithOptions`

File: services/cloudca/network_acl_rule.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	RuleNumber   string `json:"ruleNumber,omitempty"`
+	Cidr         string `json:"cidr,omitempty"`
+	Action       string `json:"action,omitempty"`
+	Protocol     string `json:"protocol,omitempty"`
+	StartPort    string `json:"startPort,omitempty"`
+	EndPort      string `json:"endPort,omitempty"`
+	IcmpType     string `json:"icmpType,omitempty"`
+	IcmpCode     string `json:"icmpCode,omitempty"`
+	TrafficType  string `json:"trafficType,omitempty"`
+	State        string `json:"state,omitempty"`
```
Comment (pdube) [inline services/cloudca/network_acl_rule.go]: There is state on network acl rules?

File: services/cloudca/network_acl_rule.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	result, err := networkAclRuleApi.entityService.Update(id, msg, map[string]string{})
+	if err != nil {
+		return nil, err
+	}
+	return parseNetworkAclRule(result), nil
+}
+
+func (networkAclRuleApi *NetworkAclRuleApi) Delete(id string) (bool, error) {
+	_, err := networkAclRuleApi.entityService.Delete(id, []byte{}, map[string]string{})
+	return err == nil, err
```
Comment (prollynomial) [inline services/cloudca/network_acl_rule.go]: I've seen this in a few places, but whenever we actually use `Delete`, we do `_, err := api.Delete(id)`. Do we need the success bool?

**Comment by pdube** [comment]
👍

File: services/cloudca/network_acl_rule.go (hunk 2 above)
Comment (simongodard) [inline services/cloudca/network_acl_rule.go]: ACS returns one (Active or Inactive) and CloudMC parses it. That's why it's also here, although we don't use it.

File: services/cloudca/network_acl_rule.go (hunk 1 above)
Comment (simongodard) [inline services/cloudca/network_acl_rule.go]: Will add the missing pointers for added consistency.

**Pull request merged**